File: 프로그래머스/1/12982. 예산/예산.cpp
#include <iostream>
#include <stdio.h>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> d, int budget) 
{
    sort(d.begin(), d.end());
    
    int idx = 0;
    while(budget > 0)
    {
        budget -= d[idx];
        idx++;
        if(idx == d.size())
        {
            break;
        }
    }
    
    if(budget < 0)
    {
        idx--;
    }
    
    return idx;
}

File: 프로그래머스/1/12982. 예산/예산.cs
using System;

public class Solution 
{
    public int solution(int[] d, int budget) 
    {
        Array.Sort(d);
        int sum = 0;
        int count = 0;
        for(int i=0; i<d.Length; i++)
        {
            sum += d[i];
            if(sum > budget)
            {
                break;    
            }
            count++;
        }
        return count;
    }
}